workers/unity/Packages/com.improbable.gdk.core/Dynamic/Dynamic.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker.Core;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core
+{
+    public static class Dynamic
+    {
+        public interface IHandler
+        {
+            void Accept<T>(uint componentId, Func<SchemaObject, World, T> deserialize) where T : ISpatialComponentData;
+        }
+
+        private static readonly Dictionary<uint, IDynamicInvokable> idsToComponents =
+            new Dictionary<uint, IDynamicInvokable>();
+
+        private static readonly Dictionary<Type, uint> componentsToIds = new Dictionary<Type, uint>();
+
+        static Dynamic()
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                foreach (var type in assembly.GetTypes())
+                {
+                    if (typeof(IDynamicInvokable).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (IDynamicInvokable) Activator.CreateInstance(type);
+                        idsToComponents.Add(instance.ComponentId, instance);
+                    }
+
+                    if (typeof(ISpatialComponentData).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (ISpatialComponentData) Activator.CreateInstance(type);
+                        componentsToIds.Add(type, instance.ComponentId);
+                    }
+                }
+            }
+        }
+
+        public static void ForEachComponent(IHandler handler)
+        {
+            foreach (var component in idsToComponents.Values)
```

Comment:
This allocates I think, would iterating over the map directly not be better?

Reply:
Forgot this was a map; it was a list at first. Will change.

Reply:
Actually I'm not sure this is worth changing as this is one time init code and I don't think there is a substantially better way of doing this now. You can avoid some minor things by getting an enumerator directly.

Reply:
I think they both call the direct non allocating GetEnumerator and the only difference is not calling dispose in the finally block.